SiriIntents/IntentHandlers/SendMessage/SendMessageIntentHandler.m

```diff
@@ -0,0 +1,159 @@
+// 
+// Copyright 2022 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import "SendMessageIntentHandler.h"
+#import "ContactResolver.h"
+#import "MXKAccountManager.h"
+
+@interface SendMessageIntentHandler ()
+
+@property (nonatomic) id<ContactResolving> contactResolver;
+
+/**
+ The room that is currently being used to send a message. This is to ensure a
+ strong ref is maintained on the `MXRoom` until sending has completed.
+ */
+@property (nonatomic) MXRoom *selectedRoom;
+
+@end
+
+@implementation SendMessageIntentHandler
+
+- (instancetype)init {
+    if (self = [super init]) {
+        _contactResolver = [[ContactResolver alloc] init];
```

Comment:
Do you think it makes sense to have multiple contact resolvers, or might it be an idea to create one in `IntentHandler` to pass into the initialisers?

Reply:
Thanks for the question! There are three options that I see:

- Let every intent handler have their own instance of `ContactResolver` (as is the case in this branch right now)
- Let the `IntentHandler` create one single instance and pass it to the intent handlers during initialization. The intent handlers keep a `strong` reference, so the `IntentHandler` does not need to manage the instance of `ContactResolver`
- Add a singleton property (`ContactResolver.shared`) to `ContactResolver`

What is your preference? Is there a fourth option that is even better? Thanks in advance for your input!